### PR TITLE
Nik/210 link up full completion summary

### LINKED ISSUE #210: [Fullstack] Link up Full Completion Summary Button
## Context

We want the full completion summary to show the user's full completion summary when clicked.

## Requirements

Look at the `src/screens/Home` for the Completion Summary button. We want this to redirect to a page which looks like [this](https://www.figma.com/design/DsK6uATSqnLdCW6zTM1Qdr/Brain-Exercise-Initiative-FALL24?node-id=5874-36955&t=FI5d1coKNuVhNfuy-4). We will need the to check if the user has completed the section or not, and if it is completed, we add the section summary. The components have already been implemented in `src/components`. If the section is not completed, display a component that matches the figma: "Please complete ... to view completion Summary." 

The state is currently being stored in Redux, so you can look at the HomeScreen Page and gameDetails to see how it has been done previously. I am working on the authentication error so I will probably push a change before you start on this ticket.

## Acceptance Criteria

- [x] Button redirects to a new page
- [x] Completion Summary page shows each component, math, reading, writing, trivia.
- [x] Completion Summary page shows grayed out component if it has not been finished yet.

### PR DESCRIPTION
**Pull request by nikvijay07**
[##] PR Title

Issue Number(s): #210 

What does this PR change and why?

- Completion Summary Page
- Created new components for each subject view
- Added Function to properly display time

https://github.com/user-attachments/assets/1fee8a8a-3dc2-4776-ba43-2f379290e528

### Checklist

- [ ]  Button redirects to a new page
- [ ] Completion Summary page shows each component, math, reading, writing, trivia.
- [ ] Completion Summary page shows grayed out component if it has not been finished yet.

### Critical Changes / Notes
- None

### Related PRs

- None

### Testing

- None
Enumerate steps to test the functionality of your ticket. This should include edge cases and testing of error handling, if applicable.

### CODE REVIEW

File: src/components/SubjectComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+                fontStyle: "normal",
+                fontWeight: "600",
+                paddingLeft: "3%",
+              }}
+            >
+              {title}
+            </Text>
+          </View>
+
+          {questionsCompleted > 0 ? (
```
Comment (nikvijay07) [inline src/components/SubjectComponent.tsx:91]: ended up making this a nested ternary so it is kind of hard to read.

**Review by johannesq23 — COMMENTED** [review]
lgtm. well done

**Review by johannesq23 — APPROVED** [review]
